### ISSUE TITLE
Panic: `sort_with` and `arg_sort` for dtype `null`

### ISSUE THREAD

**Issue by s-banach** [pola-rs/polars#14509]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
df = pl.DataFrame({"x": [None, None]})
df.sort("x")
"""PanicException: `sort_with` operation not supported for dtype `null`"""
```

```python
import polars as pl
df = pl.DataFrame({"x": [None, None], "y": [1, 2]})
df.sort("x")
"""PanicException: `arg_sort` operation not supported for dtype `null`"""
```

### Issue description

I guess Panics should be considered bugs.

### Expected behavior

Maybe sorting a null column is just a no-op? No opinion, just hopefully it doesn't panic.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.8
Index type:           UInt32
Platform:             Linux-4.18.0-425.19.2.el8_7.x86_64-x86_64-with-glibc2.28
Python:               3.11.3 | packaged by conda-forge | (main, Apr  6 2023, 08:57:19) [GCC 11.3.0]

----Optional dependencies----
adbc_driver_manager:  0.6.0
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               2023.6.0
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.7.2
numpy:                1.26.3
openpyxl:             3.1.2
pandas:               2.1.2
pyarrow:              14.0.1
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           2.0.16
xlsx2csv:             0.8.1
xlsxwriter:           3.1.2
```

</details>

**Comment by gusutabopb** [comment]
I just hit this today. Agree this should be a no-op, not a panic.

**Comment by gusutabopb** [comment]
This still isn't fully fixed. Sorting with multiple columns fails if there's a null dtype column is involved. Using the OP example:

```python
# This was fixed by #20135
df = pl.DataFrame({"x": [None, None], "y": [1, 2]})
df.sort("x")
```

```python
# InvalidOperationError: `arg_sort_multiple` operation not supported for dtype `null`
df = pl.DataFrame({"x": [None, None], "y": [1, 2]})
df.sort("x", "y")   # same as df.sort(pl.all())
```

```python
# PanicException: `into_total_ord_inner` operation not supported for dtype `null`
df = pl.DataFrame({"x": [None, None], "y": [1, 2]})
df.sort("y", "x")
```

I believe this issue should be re-opened.
